docs: add essentials section for Transactions (#1961)
FuelLabs/fuels-ts: apps/docs/src/guide/transactions/index.md

# Transactions

> A transaction is any interaction with the blockchain. This is the primary way of interacting with the Fuel Network and can include actions like transferring assets, deploying contracts and minting tokens.

Reviewer: Not all interactions with the blockchain are transactions, e.g. `provider.operations.getChain()`. Nothing comes to mind for rephrasing it, though.

Author: Hmmm interesting, I'd originally taken that from [here](https://docs.fuel.network/docs/intro/glossary/#transaction).

"A transaction is a way of interacting with the blockchain"?

Author: https://github.com/FuelLabs/fuels-ts/pull/1961/commits/5fcb542f03019c177c4234ea07a05c343dddd034

Within the Fuel Network, we have the following transaction types:

- Script
- Create
- Mint

> The SDK provides class helpers for handling script and create transactions, `ScriptTransactionRequest` and `CreateTransactionRequest` respectively. Mint transactions are recognized by the SDK but can only be created by the block producer so do not have use any use outside of block creation. Therefore there is no need to manually create them.

Reviewer: ```suggestion
The SDK provides class helpers for handling script and create transactions: `ScriptTransactionRequest` and `CreateTransactionRequest`, respectively. 
Mint transactions can only be created by the block producer and do not have any use outside of block creation. Therefore, the SDK only provides the ability to decode them.
```

If this rewording doesn't go through, there is an additional "use" in the original wording that should be removed.

> This guide will discuss how to modify transactions, set their transaction parameters and policies, and submit them to the network to then produce an extractable transaction response.

Reviewer: ```suggestion
This guide will discuss how to create, modify and submit transactions to the network, as well as how to get info on transactions that have previously executed.
```

# Transactions

> A transaction is a way of interacting with Fuel Network's blockchain and can include actions like transferring assets, deploying contracts and minting tokens.

Reviewer 2: Feel like we should prefer using this wording over '_the_ xyz blockchain' because we are a VM-focused company, and not an L2 necessarily. 

```suggestion
A transaction is a way of interacting with a Fuel blockchain and can include actions like transferring assets, deploying contracts and minting tokens.
```

Reviewer 2: Our SDK is supposed to work for _any_ blockchain using the FuelVM and not just some one chain that Fuel would deploy.

> Within the Fuel Network, we have the following transaction types:

Reviewer 2: ```suggestion
Within Fuel, we have the following transaction types:
```

Same as above. Reads better, and also more aligned with our narrative imo.

- Script
- Create
- Mint

The SDK provides class helpers for handling script and create transactions: `ScriptTransactionRequest` and `CreateTransactionRequest`, respectively.

> Mint transactions can only be created by the block producer and do not have any use outside of block creation. Therefore, the SDK only provides the ability to decode them.

Reviewer 2: Feel like this paragraph should be:

1. A _note_ rather than a main paragraph
2. Below the paragraph which is currently the last on this page

Reviewer 2: I feel this way because this paragraph contains informationa about more advanced concepts like customt script transactions etc. We should tell users about the 'straightforward' transactions first, which is what the last paragraph does.

Author: Made some changes [here](https://github.com/FuelLabs/fuels-ts/pull/1961/commits/7b20becfd6431fc1f8ff726d812551ec6b66c2ff) based on these comments.
